packages/core/contracts/common/implementation/Stakeable.sol

```diff
@@ -0,0 +1,50 @@
+/**
+ * Stakeable contract.
+ */
+
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../../oracle/interfaces/StakerInterface.sol";
+import "./Withdrawable.sol";
+
+/**
+ * @title Base contract that extends the Withdrawable contract enabling a specific role to stake ERC20 tokens against the
+ * Voting contract. Voting contract is fed in as a param rather than fetched from the finder to enable upgradability.
+ */
+abstract contract Stakeable is Withdrawable {
+    using SafeERC20 for IERC20;
+
+    uint256 private roleId;
```

Comment:
Where is this value set? Who is it supposed to be set to? The owner or the designated voter?

Reply:
it's set implicitly via the `Withdrawable.sol` contract (the existing DesignatedVoting contract that lets us set this. I'm going to re-work this to make it more explicit.